### PR TITLE
Feature/profiler mvp

### PR DESCRIPTION
**Pull request by MaksymilianDemitraszek**

### CODE REVIEW

File: protostar/starknet/cheatable_execute_entry_point.py [hunk 1]
```diff
@@ -149,6 +156,21 @@ def _run(  # type: ignore
                 run_resources=tx_execution_context.run_resources,
                 verify_secure=True,
             )
+            CheatableExecuteEntryPoint.callstack -= 1
+            if CheatableExecuteEntryPoint.callstack == 0:
+                runner.relocate()
+                try:
+                    save_profile(
```
Comment (MaksymilianDemitraszek) [inline protostar/starknet/cheatable_execute_entry_point.py:163]: I will leave dirty save like this for now, I will improve it later in separate PR

File: protostar/starknet/cheatable_execute_entry_point.py [hunk 2]
```diff
@@ -43,6 +47,8 @@
 # pylint: disable=too-many-statements
 class CheatableExecuteEntryPoint(ExecuteEntryPoint):
     cheatcode_factory: Optional["CheatcodeFactory"] = None
+    custom_hint_locals: Optional[List[HintLocal]] = None
```
Comment (mkaput) [inline protostar/starknet/cheatable_execute_entry_point.py:50]: this field was removed in #677

**Review by mkaput — COMMENTED** [review]
Please provide a more human readable PR name :)

**Review by mkaput — COMMENTED** [review]
This PR seems like it's still a draft

**Comment by MaksymilianDemitraszek** [comment]
@mkaput Yup it is kinda a draft, I opened it to give you general idea of the stucture

File: tests/integration/profile_tests/example_profile_test.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+)
+
+
+@pytest.mark.asyncio
+@pytest.mark.skip
+async def test_testing_output(run_cairo_test_runner: RunCairoTestRunnerFixture):
+    await run_cairo_test_runner(
+        Path(__file__).parent / "example_profile_contract_test.cairo"
+    )
+    assert False
```
[the remark this answers is not in the document]
Comment (MaksymilianDemitraszek) [reply tests/integration/profile_tests/example_profile_test.py:16]: I use it as a script and skip it on CI, it technically should be a script but it was easier to just use fixtures

**Pull request merged**